### ISSUE TITLE
Unable to install web extension

### ISSUE THREAD

**Issue by alcarney** [microsoft/vscode#157028]
Does this issue occur when all extensions are disabled?: N/A

- VS Code Version:  1.69.2
- OS Version: Web
- Browser: Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0
  (I have also replicated this on iOS and Edge)

Steps to Reproduce:

1. Open `vscode.dev`
2. Search for "Esbonio" in the marketplace
3. Click "Install"

![image](https://user-images.githubusercontent.com/2675694/182682148-eac26705-9f35-45a8-b2c4-e8ae647468c5.png)

I (thought I'd) recently added web extension support to my VSCode extension [Esbonio](https://marketplace.visualstudio.com/items?itemName=swyddfa.esbonio).

Before publishing the extension I tested it in a browser by following [this guide](https://code.visualstudio.com/api/extension-guides/web-extensions#test-your-web-extension-in-on-vscode.dev) and everything seemed to work as expected.

The VSCode UI also seems to agree with me that this is a valid web extension as there is no warning icon and an Install button is presented - however clicking install fails with the dialog saying it is not a valid web extension (as shown in the screenshot above).

If I had to guess what was wrong it would be the fact the extension is also an extension pack that depends on an extension that is not a valid web extension. But there is nothing in the UI that tells me that is the case and it seems to contradict the [web extension documentation](https://code.visualstudio.com/docs/editor/vscode-web#_extensions) which says the user should still be able to install it anyway.

Any help in diagnosing this would be appreciated :smile:

**Comment by alcarney** [comment]
**Note:** I've pushed an update removing the `extensionPack` field from the extension's `package.json` and it now installs correctly, so this issue does not affect the Esbonio extension anymore.

However, I did some digging and I think I found what was causing the behavior I was seeing above

https://github.com/microsoft/vscode/blob/b3dc3301dd22d9695d5e5bdc69e5e23a6db1be58/src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts#L428-L439

It looks like installation is blocked if every extension declared in an extension pack has no web support - which makes sense if the parent extension doesn't provide any features of its own.

Unfortunately, it also prevents extensions like Esbonio which also provide their own features from being installed as [this banner](https://code.visualstudio.com/assets/docs/editor/vscode-web/python-extension-limit.png) is never shown to the user.
